pydantic/types.py

```diff
@@ -156,11 +149,12 @@ class ConstrainedStr(str):
     max_length: OptionalInt = None
     curtail_length: OptionalInt = None
     regex: Optional[Pattern[str]] = None
+    strict = False
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
         yield not_none_validator
-        yield str_validator
+        yield make_arbitrary_type_validator(str) if cls.strict else str_validator
```

Comment:
does this work if you pass it an enum that inherits from `str`? Perhaps it should, but it will likely confuse many people since the `__str__` method is different from normal I think.

We should at least add a note to the docs.

Reply:
Naturally this should pass with all subclasses as we only make the exception for int's with boolean as that is a weird quirk of Python where Guido probably looked at C too much.

I can add the note to the docs.